src/Repeater/RepeaterServer.ts

```diff
@@ -0,0 +1,77 @@
+import { Protocol } from '../RequestExecutor';
+
+export interface RepeaterServerDeployedEvent {
+  repeaterId: string;
+}
+
+export interface RepeaterServerRequestEvent {
+  protocol: Protocol;
+  url: string;
+  method?: string;
+  headers?: Record<string, string | string[]>;
+  correlationIdRegex?: string;
+  body?: string;
+}
+
+export type RepeaterServerRequestResponse =
+  | {
+      protocol: Protocol;
+      statusCode?: number;
+      message?: string;
+      errorCode?: string;
+      headers?: Record<string, string | string[] | undefined>;
+      body?: string;
+    }
+  | {
+      protocol: Protocol;
+      message?: string;
+      errorCode?: string;
+    };
+
+export interface RepeaterServerReconnectionFailedEvent {
+  error: Error;
+}
+
+export interface RepeaterServerReconnectionAttemptedEvent {
+  attempt: number;
+  maxAttempts: number;
+}
+
+export interface RepeaterServerErrorEvent {
+  message: string;
+}
+
+export type RepeaterServerEventHandler<P, R = void> = P extends undefined
+  ? () => R | Promise<R>
+  : (payload: P) => R | Promise<R>;
```

Comment:
The event listeners have nothing in common with each other. Let's inline their types to make it more cohesive.

Additionally, if you're not interested in the return value, declare the return type as `unknown`, this is the more user-friendly approach. Also, get rid of redundant wrappers, such `RepeaterServerReconnectionFailedEvent`

For details please refer to the example below:

```ts
requestReceived(
    handler: (request: RepeaterServerRequestEvent) => RepeaterServerRequestResponse | Promise< RepeaterServerRequestResponse >
  ): void;

  reconnectionFailed(
    handler: (error: Error) => unknown
  ): void;
```

Reply:
Done.

But I preserved `void` type for handlers which are not supposed to return anything. It seems to be more clear from handler's contract perspective. If I used `unknown` it would mean that handlers expect some data to be returned, that further will be processed. But with `void` it clearly says that no data is needed to be returned

Reply:
> If I used unknown it would mean that handlers expect some data to be returned, that further will be processed. 

You already handle the never type silently, hiding these details from the public API by using the `void` type. 

Avoid using the `void` return type for callbacks when the return value will be ignored or handled (e.g. the `never` type). Instead, use `unknown` to explain clearly that you are not interested in the return value, regardless of what it might be, and gracefully handle all cases.  

Anyway, you can keep it as for now, since this is part of internal API.

Reply:
> Avoid using the void return type for callbacks when the return value will be ignored or handled (e.g. the never type). Instead, use unknown to explain clearly that you are not interested in the return value, regardless of what it might be, and gracefully handle all cases.

I think I don't fully understand what benefits `unknown` gives us in such cases. 

`unknown` exactly says that i'm interested in returned value, but not in it's type. `void` says that it doesn't expect any value to be returned, so it won't be processed even if returned

Reply:
> , so it won't be processed even if returned

However, you do. You handle the never, is not it so?

Reply:
These are not implementation details. Such "details" represent the post- and pre-conditions of any interface.  Using the `void` here could potentially result in incorrect inheritance, contract and LSP violations.